packages/authentication/src/providers/auth-strategy.provider.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingScope, Getter, inject} from '@loopback/context';
+import {extensionPoint, extensions, Provider} from '@loopback/core';
+import {AuthenticationMetadata} from '../decorators/authenticate.decorator';
+import {AuthenticationBindings} from '../keys';
+import {
+  AuthenticationStrategy,
+  AUTHENTICATION_STRATEGY_NOT_FOUND,
+} from '../types';
+
+//this needs to be transient, e.g. for request level context.
+@extensionPoint(
+  AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME,
+  {scope: BindingScope.TRANSIENT},
+)
+export class AuthenticationStrategyProvider
```

Comment:
Since this is part of the source code, it'd be great to add TSDocs to here for our API docs.